examples/keras_recipes/melanoma_classification_with_tfrecords.py

```diff
@@ -0,0 +1,336 @@
+"""
+Title: How to train a Keras model on TFRecord files
+Author: Amy MiHyun Jang
+Date created: 2020/07/29
+Last modified: 2020/08/06
+Description: Loading TFRecords for computer vision models
+"""
+"""
+## Introduction + Set Up
+
+TFRecords store a sequence of binary records, read linearly. They are useful format for
+storing data because they can be read efficiently. We'll explore how we can easily load
+in TFRecords for our melanoma classifier.
+"""
+
+"""shell
+! pip install gcsfs -q
+"""
+
+import re
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from functools import partial
+import matplotlib.pyplot as plt
+
+try:
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver()
+    print("Device:", tpu.master())
+    tf.config.experimental_connect_to_cluster(tpu)
+    tf.tpu.experimental.initialize_tpu_system(tpu)
+    strategy = tf.distribute.experimental.TPUStrategy(tpu)
+except:
+    strategy = tf.distribute.get_strategy()
+print("Number of replicas:", strategy.num_replicas_in_sync)
+
+"""
+Running the following cell will save time in loading the data as it allows for parallel
+processing. TPU can only work with Google Cloud files and cannot be run with local files.
+We want a bigger batch size as our data is not balanced.
+
+"""
+
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+GCS_PATH = "gs://kds-f809fea39df606e89617cd8d8e4cacc083fb6e176982f0e5d69215a8"
+BATCH_SIZE = 16 * strategy.num_replicas_in_sync
+IMAGE_SIZE = [1024, 1024]
+
+"""
+## Load the data
+
+Now we will load in our data. For this notebook, we will be importing the TFRecord files.
+It is good practice to divide the training set data into two. The smaller dataset will be
+the validation set. Having a validation set is useful to slow down overfitting.
+"""
+
+FILENAMES = tf.io.gfile.glob(GCS_PATH + "/tfrecords/train*.tfrec")
+split_ind = int(0.9 * len(FILENAMES))
+TRAINING_FILENAMES, VALID_FILENAMES = FILENAMES[:split_ind], FILENAMES[split_ind:]
+
+TEST_FILENAMES = tf.io.gfile.glob(GCS_PATH + "/tfrecords/test*.tfrec")
+print("Train TFRecord Files:", len(TRAINING_FILENAMES))
+print("Validation TFRecord Files:", len(VALID_FILENAMES))
+print("Test TFRecord Files:", len(TEST_FILENAMES))
+
+"""
+### Decoding the data
+
+The images have to be converted to tensors so that it will be a valid input in our model.
+As images utilize an RBG scale, we specify 3 channels.
+
+It is also best practice to normalize data before it is is fed into the model. For our
+image data, we will scale it down so that the value of each pixel will range from [0, 1]
+instead of [0, 255].
+
+We also reshape our data so that all of the images will be the same shape. Although the
+TFRecord files have already been reshaped for us, it is best practice to reshape the
+input so that we know exactly what's going in to our model.
+"""
+
+
+def decode_image(image):
+    image = tf.image.decode_jpeg(image, channels=3)
+    image = tf.cast(image, tf.float32) / 255.0
```

Comment:
Since you're using Xception, you should do this via `applications.xception.preprocess_input`. Indeed, different applications use different value normalization scheme, so it's better to always import the one that corresponds to the model you're using.